### DISCUSSION TITLE
Page API: ODataKeyFields

### DISCUSSION THREAD

**Discussion by waldo1001** [StefanMaron/BusinessCentral.LinterCop#45] [New Rule]
May be a warning for when people forget the ODataKeyFields-property on a page-based API?

**Comment by jwikman** [comment]
And I think that the rule should include that this property should always be set to `SystemId`, as specified at https://alguidelines.dev/docs/bestpractices/api-page/#odatakeyfields.

**Comment by Arthurvdv** [comment]
The [v0.30.28](https://github.com/StefanMaron/BusinessCentral.LinterCop/releases/tag/v0.30.28) release of the LinterCop has now multiple rules based on the [Best Practices for API pages](https://alguidelines.dev/docs/bestpractices/api-page/) from the alguidelines.dev.

* [LC0061 - Pages of type API must have the ODataKeyFields property set to the SystemId field.](https://github.com/StefanMaron/BusinessCentral.LinterCop/wiki/LC0061)
* [LC0062 - Mandatory field is missing on API page.](https://github.com/StefanMaron/BusinessCentral.LinterCop/wiki/LC0062)
* [LC0063 - Consider naming field with a more descriptive name.](https://github.com/StefanMaron/BusinessCentral.LinterCop/wiki/LC0063)
